### PR TITLE
Add default value of Glue GetPartitions MaxResults

### LINKED ISSUE #2996: Add support for MaxResults on Glue Hive Metastore
I was tracing our Presto cluster with AWS CloudTrail and it issues a lot of GetPartitions (sometimes hundreds) for a table. We are using AWS Glue.

My first question is where is the MaxResults set to Presto?

https://docs.aws.amazon.com/glue/latest/webapi/API_GetPartitions.html

```
MaxResults
The maximum number of partitions to return in a single response.
Type: Integer
Valid Range: Minimum value of 1. Maximum value of 1000.
Required: No
```

### PR DESCRIPTION
**Pull request by l1x**
Fixes #2996

As per our discussion with @findepi and @ebyhr here is the fix for using batched requests with AWS Glue.

### CODE REVIEW

**Comment by l1x** [comment]
Discussed in: https://github.com/prestosql/presto/issues/2996

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueHiveMetastore.java [hunk 1]
```diff
@@ -147,6 +147,7 @@
     private static final String WILDCARD_EXPRESSION = "";
     private static final int BATCH_GET_PARTITION_MAX_PAGE_SIZE = 1000;
     private static final int BATCH_CREATE_PARTITION_MAX_PAGE_SIZE = 100;
+    private static final int AWS_GLUE_GET_PARTITIONS_MAX_RESULTS = 128;
```
Comment (sopel39) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueHiveMetastore.java]: why 128 and not some other value (e.g larger)?

Comment (l1x) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueHiveMetastore.java]: Ultimately we need to find a balance between how much data you transfer at once vs how many iterations do you need on average for a typical use case. 128 is much larger than the current default (1) so I would start with this. I also think that we could make this configurable. What would you like to see as the default value instead?

Comment (sopel39) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueHiveMetastore.java]: Is there some value that @l1x recommends?

Comment (l1x) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueHiveMetastore.java]: I guess 128.

Comment (sopel39) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueHiveMetastore.java]: haha. I meant what AWS recommends?

**Comment by hashhar** [comment]
Thanks a lot @l1x. We are personally hit hard by this issue since we want to move to Presto from Athena but the planning performance was abysmal for tables with significant number of partitions (>10k). Just wanted to let you know I appreciate this.

PS: I came here when I myself was digging into this issue and came to the same conclusion that you did.

**Comment by hashhar** [comment]
AWS doesn't recommend any value but in practice from a small script I made to test I can see that there are no downsides to setting this to the max of 1000. The total time with a smaller value like 128 vs the max value is appreciable (3x slower for fetching almost 8k partitions). You can simulate this in a very crude way using the `aws-cli` command.

```sh
aws glue get-partitions --database-name db --table-name table --expression "dt>='2020-02-01'" --page-size 128
aws glue get-partitions --database-name db --table-name table --expression "dt>='2020-02-01'" --page-size 1000
```
The aws cli internally implements the `nextToken` loop so that all results are returned.

**Comment by l1x** [comment]
@hashhar I am glad I could help! Let me know how it goes. We could bump the number even higher or make it configurable.

**Comment by hashhar** [comment]
@l1x We are now running PrestoSQL on production and it haven't seen any practical issues with the smaller page size for queries scanning upto 50k partitions. Will keep an eye out.

**Pull request merged**